_posts/2021-02-22-understanding-non-key-joins-with-quarkus-kafka-streams-extension.adoc

```diff
@@ -0,0 +1,443 @@
+which you can find in the https://github.com/debezium/debezium-examples/tree/master/kstreams-fk-join[debezium-examples] repo.
+
+To launch all the containers, also building the aggregator container image, run the the following:
+
+----
+export DEBEZIUM_VERSION=1.4
+
+docker-compose up --build
+----
+
+To register the Debezium Connector with Kafka Connect,
+you need specify the configuration properties like name of the connector, database hostname, user, password, port, name of the database, etc.
+Create a file https://github.com/debezium/debezium-examples/blob/master/kstreams-fk-join/register-postgres.json[register-postgres.json] with the following contents:
+
+[source,json]
+----
+{
+    "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+    "tasks.max": "1",
+    "database.hostname": "postgres",
+    "database.port": "5432",
+    "database.user": "postgres",
+    "database.password": "postgres",
+    "database.dbname" : "postgres",
+    "database.server.name": "dbserver1",
+    "schema.include": "inventory",
+    "decimal.handling.mode" : "string",
+    "key.converter": "org.apache.kafka.connect.json.JsonConverter",
+    "key.converter.schemas.enable": "false",
+    "value.converter": "org.apache.kafka.connect.json.JsonConverter",
+    "value.converter.schemas.enable": "false"
+}
+----
+
+Configure the Debezium Connector:
+
+----
+http PUT http://localhost:8083/connectors/inventory-connector/config < register-postgres.json
+----
+
+Now run an instance of the `debezium/tooling` image:
+
+----
+docker run --tty --rm \
+    --network kstreams-fk-join-network \
+    debezium/tooling:1.1 \
+----
+
+This image provides several useful tools such as https://github.com/edenhill/kafkacat[kafkacat]. Within the tooling container, run kafkacat to examine the results of the streaming pipeline:
+
+----
+kafkacat -b kafka:9092 -C -o beginning -q \
+    -t customers-with-addresses | jq .
+----
+
+== Running Natively
+Kafka Streams applications can easily be scaled out i.e. the load will be shared amongst multiple instances of the application, each processing just a subset of the partitions of the input topic(s).
+"Running Quarkus applications in native code via GraalVM has a very fast start-up time, the applications use significantly less memory when compiled to native code. This allows you to start as many instances of a Kafka Streams pipeline in parallel in a very memory-efficient way".
+If you want to run this application in `native` mode, set the `QUARKUS_MODE` as `native` and run:
+
+----
+mvn clean package -Pnative
+----
+
+To learn more about running Kafka Streams applications as a native binary, please refer to the https://quarkus.io/guides/kafka-streams#running-natively[reference guide].
+
+== More Insights on the Kafka Streams Extension
```

Comment:
Are in this section any sentences remaining which need proper quoting?

Reply:
I have done few modifications. Should I push a separate commit?

Reply:
Yes, please. If you haven't done so, you'll need to pull my commit first to your local fork before apply your changes. Just let me know, if you need help with that.